src/main/kotlin/biokotlin/cli/MutateProteins.kt

```diff
@@ -0,0 +1,380 @@
+package biokotlin.cli
+
+import biokotlin.seq.ProteinSeqRecord
+import biokotlin.seqIO.FastaIO
+import biokotlin.seqIO.SeqType
+import biokotlin.util.bufferedReader
+import biokotlin.util.bufferedWriter
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.boolean
+import com.github.ajalt.clikt.parameters.types.enum
+import com.github.ajalt.clikt.parameters.types.int
+import com.google.common.collect.HashMultimap
+import com.google.common.collect.Multimap
+import org.apache.logging.log4j.LogManager
+import java.io.BufferedWriter
+import kotlin.random.Random
+
+/**
+ * This command randomly mutates the protein sequences in a fasta file.
+ * The mutations can be deletions, point mutations, or insertions.
+ * The output is written to a new fasta file.
+ * The mutations can be put in or out of ranges defined by a bedfile.
+ */
+class MutateProteins : CliktCommand(help = "Mutate Proteins") {
+
+    private val myLogger = LogManager.getLogger(MutateProteins::class.java)
+
+    enum class TypeMutation {
+        DELETION, POINT_MUTATION, INSERTION
+    }
+
+    private val proteinLetters = arrayOf(
+        'R', 'H', 'K', 'D', 'E', 'S', 'T', 'N', 'Q', 'C', 'G', 'P', 'A', 'V', 'I', 'L', 'M', 'F', 'Y', 'W'
+    )
+
+    private val numProteinLetters = proteinLetters.size
+
+    val inputFasta by option(help = "Full path to input fasta file")
+        .required()
+
+    val outputFasta by option(help = "Full path to output fasta file")
+        .required()
+
+    val bedfile by option(help = "Full path to bedfile")
+        .required()
+
+    val typeMutation by option(help = "Type of mutation")
+        .enum<TypeMutation>()
+        .default(TypeMutation.POINT_MUTATION)
+
+    val putMutationsInRanges by option(help = "Put mutations in ranges defined by bedfile")
+        .boolean()
+        .default(true)
+
+    val length by option(help = "Length of the deletion mutation")
+        .int()
+        .default(5)
+
+    val numMutations by option(help = "Number of point mutations")
+        .int()
+        .default(10)
+
+    val mutatedIndicesBedfile by option(help = "Full path to bedfile to output mutated indices")
+
+    val randomSeed by option(help = "Random seed")
+        .int()
+        .default(1234)
+
+    override fun run() {
```

Comment:
There is a lot of extra code in this command.  Can some of it move over to the Biokotlin side of things?

Reply:
Yes I think so. But I need to have another task to remove the Clikt stuff from BioKotlin. And do that sort of refactoring there. Which would allow me to refactor the stuff in biokotlin-tools.  I suggest letting this go in like this for now.